### PR TITLE
159 added map dialog to new/edit benificiary

### LINKED ISSUE #159: [Beneficiary] Find the coordinate for the address using the map
When click on map icon or the address textbox, show the map, with the possibility to add the address. When clicked on the blue button from the bottom left of the popup, add the address to the textbox, and use the latitude and longitude in the json sent to the backend.

the https://dev.voluntar.md/admin/volunteers/list page already has the map functionality. use it if possible.

https://www.figma.com/file/oCN3NSECKQnS4PWhZnWGCT/Dashboard_Voluntar.md_V2.0?node-id=820%3A1735

image

image

### PR DESCRIPTION
**Pull request by valeraf**
Closes GH-159
added map dialog to new/edit benificiary, to get lat/long and address autocomplete

### CODE REVIEW

File: src/app/admin/beneficiaries/beneficiaries-list/beneficiaries-list.component.scss [hunk 1]
```diff
@@ -82,3 +82,17 @@ h1 {
     }
   }
 }
+
+mat-select {
+  ::ng-deep {
```
Comment (unrevised6419) [inline src/app/admin/beneficiaries/beneficiaries-list/beneficiaries-list.component.scss:87]: Can we use something different then `ng-deep`? It is deprecated.

https://angular.io/guide/deprecations#deep--and-ng-deep-component-style-selectors

Comment (BaxLi) [reply src/app/admin/beneficiaries/beneficiaries-list/beneficiaries-list.component.scss:87]: one that Angular actually recommends, and that’s prepending any ng-deep rule with the :host modifier
Example - styles : [':host ::ng-deep h1 { color:red; }']

Comment (johnsmithm) [reply src/app/admin/beneficiaries/beneficiaries-list/beneficiaries-list.component.scss:87]: @valeraf

**Comment by unrevised6419** [comment]
Because this PR is old and has some conflicts I'll close it, in favor of another one that I will open tomorrow with credits to this one.

**Pull request closed without merging**